activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -90,6 +98,11 @@ def commit_records
         ite.each do |i|
           i.committed!(should_run_callbacks: false)
         end
+        @records.uniq.each do |i|
+          if i.weakref_alive?
+            i.committed!(should_run_callbacks: false)
+          end
```

Comment:
`committed!` method do 2 things. Restore state on the record, and call de callback. In here I am only restoring the state as I am passing `should_run_callbacks: false`. 
That said. Yes only non-gc records will receive that call, as those are the one we care, as someone else is holding a reference to them, so it needs to have the latest state.